### PR TITLE
Make the output from Dulwich command class behave like Git

### PR DESCRIPTION
**Pull request by hsorby**

### CODE REVIEW

**Comment by metatoaster** [comment]
Given that we are unlikely to bring back support for Mercurial ever, we can probably drop the Mercurial related classes and tests to make this pass CI.

**Comment by hsorby** [comment]
Sounds good to me.

**Comment by hsorby** [comment]
Since we don't really worry about Mercurial I'll try a quick fix first.

File: src/pmr2/wfctrl/cmd.py [hunk 1]
```diff
@@ -183,34 +190,41 @@ def read_remote(self, workspace, target_remote=None, **kw):
...
             args.append('--all')
         elif isinstance(branches, list):  # pragma: no cover
             args.extend(branches)
 
-        return self.execute(*args)
+        output = self.execute(*args)
+        return_code = 0
+        if output[1]:
+            return_code = 1
+        return '\n'.join(output[0]).encode(), output[1], return_code
```
Comment (metatoaster) [inline src/pmr2/wfctrl/cmd.py:227]: The `Popen` return code is captured in the third element (idx 2) of the tuple as per https://github.com/PMR2/pmr2.wfctrl/blob/master/src/pmr2/wfctrl/core.py#L304-L305  - just use that instead?  Avoids the condition check and thus the coverage issue.

Comment (hsorby) [reply src/pmr2/wfctrl/cmd.py]: Done.

**Comment by metatoaster** [comment]
Actually went to the core of the problem after I realized these workarounds don't make sense - the problem being the test set up helpers weren't updated to return the 3-tuple.  This was fixed so the nonsense handling no need to happen.  I pushed to my own fork and will be merging this in shortly after CI is done for that.

**Pull request merged**